src/main/java/com/offz/spigot/staminaclimb/stamina/StaminaTask.kt

```diff
@@ -6,53 +6,107 @@ import com.offz.spigot.staminaclimb.climbing.ClimbBehaviour
 import com.offz.spigot.staminaclimb.config.StaminaConfig
 import org.bukkit.Bukkit
 import org.bukkit.GameMode
+import org.bukkit.util.Vector
+import org.bukkit.Location
+import org.bukkit.entity.Player
 import org.bukkit.boss.BarColor
 import org.bukkit.potion.PotionEffect
 import org.bukkit.potion.PotionEffectType
 import org.bukkit.scheduler.BukkitRunnable
+import java.util.Calendar
+import java.util.UUID
+import kotlin.math.abs
 
 class StaminaTask : BukkitRunnable() {
+
+    private var previousTickTime: Long = 0;
+    private val targetDeltaTime = 1.0 / 20.0
+
+    // Not sure how to get the World to make a Location so I'm just doing this | TODO: change to a Location
+    private var prevX = 0.0
+    private var prevY = 0.0
+    private var prevZ = 0.0
+
+    private var ticksSinceLastColorSwitch = 0
```

Comment:
This also can't be stored here and should be done per player. We should probably just make a map of uuid to player data object instead of having a bunch of maps. Actually I forget if I already did that or not but I'm pretty sure I was too stupid for that when I coded this...